slither/core/slither_core.py

```diff
@@ -265,13 +265,45 @@ def print_functions(self, d: str):
     ###################################################################################
     ###################################################################################
 
+    def has_ignore_comment(self, r: Dict) -> bool:
+        """
+        Check if the result has an ignore comment on the proceeding line, in which case, it is not valid
+        """
+        mapping_elements_with_lines = [
+            dict(
+                file=os.path.normpath(elem["source_mapping"]["filename_absolute"]),
+                lines=elem["source_mapping"]["lines"],
+            )
+            for elem in r["elements"]
+            if "source_mapping" in elem
+            and "filename_absolute" in elem["source_mapping"]
+            and "lines" in elem["source_mapping"]
+            and len(elem["source_mapping"]["lines"]) > 0
+        ]
+
+        for ele in mapping_elements_with_lines:
+            ignore_line_index = min(ele["lines"]) - 1
+            code_lines = self.source_code[ele["file"]].splitlines()
+            if len(code_lines) < ignore_line_index + 1 or ignore_line_index - 1 < 0:
+                return False
+            ignore_line_text = code_lines[ignore_line_index - 1]
+            match = re.match(r"^\s*//\s*slither-disable-next-line\s*", ignore_line_text)
```

Comment:
![image](https://user-images.githubusercontent.com/10009354/101771564-6f5c2480-3aea-11eb-931c-226f5b7d38e2.png)
Leaving it here in case someone wonders what this regex matches (via https://regexper.com/#%5E%5Cs*%5C%2F%5C%2F%5Cs*slither-disable-next-line%5Cs*)

Reply:
Since we match from "Start of line" and then allow only for white spaces the suppress detection feature won't work if the comment is made in a line after code.

Reply:
If we want to support comments made on a line after code, I don't think